### PR TITLE
Added parameters to module classes

### PR DESCRIPTION
**Pull request by annando**
While working at the "pinned" functionality I realized that we do generate a lot of warnings when the parameter variable isn't added to all functions.

### CODE REVIEW

**Comment by annando** [comment]
@MrPetovan done. The only change in functionality in this PR is the "starred" class, since I there tested the things that I wanted to apply to the "pinned" class.

File: src/BaseModule.php [hunk 1]
```diff
@@ -60,20 +58,17 @@ public static function content()
 	 * Extend this method if the module is supposed to process POST requests.
 	 * Doesn't display any content
 	 */
-	public static function post()
+	public static function post(array $parameters = [])
 	{
-		// $a = self::getApp();
```
Comment (MrPetovan) [inline src/BaseModule.php]: Please don't remove these comments meant to display what `post()` is supposed to end with.

Comment (annando) [reply src/BaseModule.php]: Ah, I thought these had been stuff that had been left over.

File: src/Module/Starred.php [hunk 2]
```diff
@@ -10,51 +10,36 @@
...
-			exit();
+		if (empty($parameters['item'])) {
+			exit;
+		} else {
+			$itemId = intval($parameters['item']);
 		}
 
 		$item = Item::selectFirstForUser(local_user(), ['starred'], ['uid' => local_user(), 'id' => $itemId]);
 		if (empty($item)) {
 			exit();
```
Comment (MrPetovan) [inline src/Module/Starred.php]: Please throw HTTPExceptions instead of exiting.

File: src/Module/Starred.php [hunk 3]
```diff
@@ -10,51 +10,36 @@
...
-			$a->internalRedirect($returnPath . $rand);
+		if (!empty($returnPath)) {
+			$rand = '_=' . time() . (strpos($returnPath, '?') ? '&' : '?') . 'rand';
+			self::getApp()->internalRedirect($returnPath . $rand);
 		}
 
 		// the json doesn't really matter, it will either be 0 or 1
-		echo json_encode($starred);
+		echo json_encode((int)$starred);
 		exit();
```
Comment (MrPetovan) [inline src/Module/Starred.php]: Please use `System::jsonExit` instead.

Comment (annando) [reply src/Module/Starred.php]: For some unknown reason the jsonExit hadn't worked. I'm unsure why, so I changed that back.

File: src/Module/Starred.php [hunk 4]
```diff
@@ -10,51 +10,36 @@
...
 {
-	public static function rawContent()
+	public static function rawContent(array $parameters = [])
 	{
-		$a = self::getApp();
-		$starred = 0;
-		$itemId = null;
-
 		if (!local_user()) {
 			exit();
```
[the remark this answers is not in the document]
Comment (annando) [reply src/Module/Starred.php]: I'm not sure which and how.

[the remark this answers is not in the document]
Comment (MrPetovan) [reply src/Module/Starred.php]: When it is supposed to be an authenticated call, `throw new \Friendica\Network\HTTPException\ForbiddenException(...);` or `throw new \Friendica\Network\HTTPException\UnauthorizedException(...);`

When it's a missing parameter: `throw new \Friendica\Network\HTTPException\BadRequestException(...);`

When the SQL query yields nothing: `throw new \Friendica\Network\HTTPException\NotFoundException(...);`

When it isn't supposed to fail: `throw new \Friendica\Network\HTTPException\InternalServerErrorException(...);`

File: src/Module/Starred.php (hunk 3 above)
Comment (MrPetovan) [reply src/Module/Starred.php]: But `System::jsonExit` is just `echo json_encode($data); exit;`!?

File: src/Module/Starred.php [hunk 5]
```diff
@@ -10,51 +10,36 @@
...
-	public static function rawContent()
+	public static function rawContent(array $parameters = [])
 	{
-		$a = self::getApp();
-		$starred = 0;
-		$itemId = null;
-
 		if (!local_user()) {
-			exit();
+			throw new \Friendica\Network\HTTPException\NotFoundException();
```
Comment (MrPetovan) [inline src/Module/Starred.php]: `ForbiddenException`.

File: src/Module/Starred.php [hunk 6]
```diff
@@ -10,51 +10,36 @@
...
 		}
 
-		// @TODO: Replace with parameter from router
-		if ($a->argc > 1) {
-			$itemId = intval($a->argv[1]);
+		if (empty($parameters['item'])) {
+			throw new \Friendica\Network\HTTPException\NotFoundException();
```
Comment (MrPetovan) [inline src/Module/Starred.php]: `BadRequestException`

**Comment by annando** [comment]
Changed again

**Pull request merged**